packages/headless/src/features/folding/folding-slice.test.ts

```diff
@@ -1,163 +1,144 @@
+import {AnyAction} from 'redux';
 import {buildMockResultWithFolding} from '../../test/mock-result-with-folding';
 import {buildMockSearch} from '../../test/mock-search';
 import {buildMockSearchResponse} from '../../test/mock-search-response';
-import {executeSearch, fetchMoreResults} from '../search/search-actions';
-import {foldingReducer} from './folding-slice';
-import {
-  FoldedResult,
-  FoldingState,
-  getFoldingInitialState,
-} from './folding-state';
-
-function getEncyclopediaCollection() {
-  const collection = 'encyclopedia';
-
-  const animalsResult = buildMockResultWithFolding({
-    uniqueId: '_animals',
-    raw: {urihash: '', collection, id: ['animals']},
-  });
-
-  const ballPythonsResult = buildMockResultWithFolding({
-    uniqueId: '_ballPythons',
-    raw: {urihash: '', collection, parent: 'reptiles'},
-  });
+import {executeSearch} from '../search/search-actions';
+import {foldingReducer, ResultWithFolding} from './folding-slice';
+import {FoldedResult, FoldingState} from './folding-state';
 
-  const geckosResult = buildMockResultWithFolding({
-    uniqueId: '_geckos',
-    raw: {urihash: '', collection, parent: 'reptiles'},
-  });
+interface Hierarchy {
+  name: string;
+  children?: Hierarchy[];
+}
 
-  const mammalsResult = buildMockResultWithFolding({
-    uniqueId: '_mammals',
+function buildMockResultsFromHierarchy(
+  collectionName: string,
+  root: Hierarchy,
+  parentName?: string
+): ResultWithFolding[] {
+  const result = buildMockResultWithFolding({
+    uniqueId: root.name,
+    title: root.name,
     raw: {
       urihash: '',
-      collection,
-      parent: 'animals',
+      collection: collectionName,
+      id: root.name,
     },
-    parentResult: {...animalsResult},
   });
 
-  const relevantResult = buildMockResultWithFolding({
-    uniqueId: '_reptiles',
-    raw: {
-      urihash: '',
-      collection,
-      id: ['reptiles'],
-      parent: 'animals',
-    },
-    parentResult: {...animalsResult},
-    childResults: [mammalsResult, ballPythonsResult, geckosResult],
-  });
+  if (parentName) {
+    result.raw.parent = parentName;
+  }
 
-  return {
-    animalsResult,
-    relevantResult,
-    ballPythonsResult,
-    geckosResult,
-    mammalsResult,
-  };
+  return [
+    result,
+    ...(root.children ?? []).flatMap((child) =>
+      buildMockResultsFromHierarchy(collectionName, child, root.name)
+    ),
+  ];
 }
 
-describe('folding slice', () => {
-  it('should be disabled by default', () => {
-    expect(foldingReducer(undefined, {type: ''}).enabled).toBeFalsy();
-  });
+function emulateAPIFolding(
```

Comment:
This util emulates what search API would return given a list of results with folding fields from a single collection. It's especially useful in unit tests to try testing a different "most relevant" result (the `target` parameter).